app/src/main/java/io/appium/uiautomator2/handler/ScrollToElement.java

```diff
@@ -47,17 +47,17 @@ public AppiumResponse safeHandle(IHttpRequest request) {
             try {
                 scrollElementUiObject = (UiObject) scrollToElement.getUiObject();
             } catch (Exception e) {
-                errorMsg = "Scroll to Element";
+                errorMsg.append("Scroll to Element");
             }
         } catch (Exception e) {
-            errorMsg = "Element";
+            errorMsg.append("Element");
         }
 
-        if (errorMsg != null) {
-            errorMsg += " was not an instance of UiObject; only UiSelector is supported. " +
-                        "Ensure you use the -android uiautomator2 locator strategy when " +
-                        "finding elements for use with ScrollToElement";
-            Logger.error(errorMsg);
+        if (errorMsg.toString() != "") {
```

Comment:
one should never use comparison operator for strings in Java. The optimal solution here would be errorMsg.toString().isEmpty()

Reply:
arg, this one always bites me, too used to JS :-)